uwblueprint/planet-read
# Establish homepage in default state

## Notion ticket link

[Create homepage and state](https://www.notion.so/uwblueprintexecs/Create-homepage-and-state-34433422ca004acda806a888f2711f92)



## Implementation description
* Created `Header` component with CSS. No interaction right now -> added logo and placeholder user icon assets into new `assets` directory as SVGs
* Created `ToggleButton` component with CSS. Basically a fancy checkbox, but there's a point of discussion below. Referenced https://webomnizz.com/how-to-create-a-toggle-switch-button-in-react/, not sure if I'm using all of the many `divs` and `spans` correctly
* Implemented query logic in `HomePage` component. `Enum` file in constants directory for general purpose

Changes to `StoryCard` and `StoryList`:
* added CSS (and some was brought out to parent component `HomePage`)
* changed Props so that it also includes `id/storyId` (same concept, different name depending on the table queried so aliased to `storyId` in https://github.com/uwblueprint/planet-read/blob/69ad057f2da02df2f2471cb200376ae9a6e15e08/frontend/src/components/pages/HomePage.tsx#L42) and optionally `storyTranslationId` if it's a translation. This was discussed with @jennchenn and @gaoxk on Slack. [Update: this PR was rebased on top of the recently-merged https://github.com/uwblueprint/planet-read/pull/36. Should consider everything that's happening]
* `cardId` created as a key for mapping in `StoryList` out of `storyId` and `storyTranslationId` if it exists (this forces uniqueness, `storyId` will only show up once without `storyTranslationId` and may show up multiple times with different `storyTranslationId`s. This `cardId` is not used anywhere else. 



## Steps to test
1. Set up DBs with some good test data. At some point, try with one of the queries returning an empty array to see the <NoStoryCard /> pop up. This is what my `story_translation_contents` table looked like, mostly the test data in https://github.com/uwblueprint/planet-read/pull/32 with the last entry changed to have a `reviewer_id`:
![image](https://user-images.githubusercontent.com/40974372/123026628-812c5780-d3aa-11eb-8a08-20686be51ad2.png)
2. Run everything and hop on over to `/stories`. If you're using the test data above, please log in as whichever is user 1 for you. 

For 3 and 4:
![homepage default state done](https://user-images.githubusercontent.com/40974372/123028644-ae2e3980-d3ad-11eb-8aa5-a58e3f8608ca.gif)

3. On the first visit, this should load the "My work" section with your translation stories. Check that the expected story cards are rendered.
4. Toggle over to the "Browse stories" section. Check that the expected story cards are rendered.
5. Change your role to reviewer. This has to be done manually: https://github.com/uwblueprint/planet-read/pull/35/files#r656720490 Check "My work" and "Browse stories" again.


## What should reviewers focus on?
* Left a comment in `Header.css`. You can see the scroll bar pop up and shift everything in the first gif commented below, what's the expected behaviour?
> this width may change on toggle
> This happens when the cursor bar on the right is asked to appear
> aka when My Stories has # of entries that > screen and Browse doesn't
> or vice versa

* Left a comment in `ToggleButton.tsx`. I tried to make the component re-usable, so it doesn't know details like the option names, however it _does_ default to true on the left always. Maybe this is a UX question, but is that ok?
>    How can we make it super visible that the true state is always displayed on the left
>    Or do we want this to change?
>    Right now, just specified in logic / className
>    
>    Left = checked = true
>    Right = unchecked = false

* Is anything taking on more responsibility than they should? (ie. more stateful than it should be)

* Is there any instances where I misuse `id` vs. `className` attributes? 
* There's just no error handling :yikes: -> how should I be handling errors/what errors happen?
* So much `px` size hardcoding, that's perfectly right, right?

## Checklist
